wp-includes/rest-api/endpoints/class-wp-rest-customize-panels-controller.php

```diff
@@ -0,0 +1,201 @@
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<panel>[\w-]+)', array(
+			'args' => array(
+				'panel' => array(
+					'description' => __( 'An alphanumeric identifier for the panel.' ),
+					'type'        => 'string',
+				),
+			),
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_item' ),
+				'permission_callback' => array( $this, 'get_item_permissions_check' ),
+				'args'                => array(
+					'context' => $this->get_context_param( array(
+						'default' => 'view',
+					) ),
+				),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+	}
+
+	/**
+	 * Retrieves the panel's schema, conforming to JSON Schema.
+	 *
+	 * @since 4.?.?
+	 * @access public
+	 *
+	 * @return array Item schema data.
+	 */
+	public function get_item_schema() {
+
+		$schema = array(
+			'$schema'    => 'http://json-schema.org/schema#',
+			'title'      => 'panel',
+			'type'       => 'object',
+			'properties' => array(
+				'description'     => array(
+					'description' => __( 'Panel description.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'name'            => array(
+					'description' => __( 'The title for the panel.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'priority'        => array(
+					'description' => __( 'The priority of the panel.' ),
+					'type'        => 'integer',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'sections'        => array(
+					'description' => __( 'The sections of the panel.' ),
+					'type'        => 'object',
+					'context'     => array( 'embed', 'view', 'edit' ),
+					'readonly'    => true,
+				),
+				'slug'            => array(
+					'description' => __( 'An alphanumeric identifier for the panel.' ),
+					'type'        => 'string',
+					'context'     => array( 'embed', 'view' ),
+					'readonly'    => true,
+				),
+				'theme_supports'  => array(
+					'description' => __( 'Theme features required to support the panel.' ),
+					'type'        => 'object',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
+				'type'            => array(
+					'description' => __( 'Type of the panel.' ),
+					'type'        => 'string',
+					'context'     => array( 'view' ),
+					'readonly'    => true,
+				),
```

Comment:
Maybe they would just be extra props that would be exported without definition in the schema. They would be all readonly anyway. The schema could then add `'additionalProperties' => 'true'`, at least as I understand https://spacetelescope.github.io/understanding-json-schema/reference/object.html